vfs/fs_utils.h

```diff
@@ -48,10 +48,22 @@ struct FsObjectInfo {
     std::map<std::string, Json::Value> objectMetadata;  ///< Metadata from obj
 
     operator bool () const { return exists; }
+
+    // make dynamic_pointer_cast work otherwise not detected as polymorphism
+    // struct
+    virtual ~FsObjectInfo() {}
 };
 
 DECLARE_STRUCTURE_DESCRIPTION(FsObjectInfo);
 
+struct ExcAwareObjectInfo : FsObjectInfo {
```

Comment:
Why does this need to derive from FsObjectInfo?  Can't it include that as a member instead?